pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -70,12 +63,8 @@ func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Cont
 	return azidext.NewTokenCredentialAdapter(tokenCredential, scopes), nil
 }
 
-func GetTokenCredential(environment *azureclient.AROEnvironment, credentials *Credentials) (azcore.TokenCredential, error) {
-	return azidentity.NewClientSecretCredential(
-		string(credentials.TenantID),
-		string(credentials.ClientID),
-		string(credentials.ClientSecret),
-		environment.ClientSecretCredentialOptions())
+func GetTokenCredential(environment *azureclient.AROEnvironment) (azcore.TokenCredential, error) {
+	return azidentity.NewDefaultAzureCredential(environment.DefaultAzureCredentialOptions())
 }
 
 // AzCredentials gets Cluster Service Principal credentials from the Kubernetes secrets
```

Comment:
Do we need `AzCredentials()` anymore? Browsing through the code base, It looks like we've replaced all uses of it with `GetTokenCredential()`, except for the unit test of `AzCredentials()`.

Reply:
Answered in other discussion thread about this topic.